src/js/components/InfiniteScroll/__tests__/InfiniteScroll-test.js

```diff
@@ -75,4 +80,62 @@ describe('InfiniteScroll', () => {
     );
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test(`Should render items equal to the length of 
+  step + 1 when step < items.length`, () => {
+    const step = 50;
+    const { container } = render(
+      <Grommet>
+        <InfiniteScroll
+          items={simpleItems(1000)}
+          // show={117}
+          step={step}
+        >
+          {item => <Box key={item}>{item}</Box>}
+        </InfiniteScroll>
+      </Grommet>,
+    );
+    // When step < the length of the number of items, an additional
+    // item is rendered as a marker for when scrolling nears the end
+    // of the rendered items. This is why expectedItems is step + 1.
+    const renderedItems = container.firstChild.children.length;
+    const expectedItems = step + 1;
+    expect(renderedItems).toEqual(expectedItems);
+  });
+
+  test(`Should render items equal to the length of 
+  step + 1 when step = array`, () => {
+    const step = 200;
+    const { container } = render(
+      <Grommet>
+        <InfiniteScroll items={simpleItems(200)} show={117} step={step}>
+          {item => <Box key={item}>{item}</Box>}
+        </InfiniteScroll>
+      </Grommet>,
+    );
+    // When step = the length of the number of items, an additional
+    // item is rendered as a marker for when scrolling nears the end
+    // of the rendered items. This is why expectedItems is step + 1.
+    const renderedItems = container.firstChild.children.length;
+    const expectedItems = step + 1;
+    expect(renderedItems).toEqual(expectedItems);
+  });
+
+  test(`Should render items equal to the length of 
+  step when step > array`, () => {
+    const { container } = render(
+      <Grommet>
+        <InfiniteScroll items={simpleItems(1000)} show={117} step={1050}>
+          {item => <Box key={item}>{item}</Box>}
+        </InfiniteScroll>
+      </Grommet>,
+    );
+    // When step > the length of the number of items,
+    // an additional item is NOT rendered because the items
+    // don't approach the end of the page. This is why the total
+    // number of items is the length of simpleItems, 1000.
+    const renderedItems = container.firstChild.children.length;
+    const expectedItems = 1000;
```

Comment:
What would happen if simpleItems was redefined at some point in the future?

```suggestion
    const expectedItems = simpleItems.length;
```